src/request.js

```diff
@@ -1,34 +1,77 @@
+// @flow
+
 const EventEmitter = require('events').EventEmitter;
 const TYPES = require('./data-type').typeByName;
 const RequestError = require('./errors').RequestError;
 
+// TODO: Figure out how to type the `rows` parameter here.
+type CompletionCallback = (error: ?Error, rowCount: ?number, rows: any) => void;
+
+type Parameter = {
+  // TODO: `type` must be a valid TDS value type
+  type: any,
+  name: string,
+  value: mixed,
+  output: boolean,
+  length: ?number,
+  precision: ?number,
+  scale: ?number
+};
+
+type ParameterOptions = {
+  output?: boolean,
+  length?: number,
+  precision?: number,
+  scale?: number
+}
+
 module.exports = class Request extends EventEmitter {
-  constructor(sqlTextOrProcedure, callback) {
+  sqlTextOrProcedure: ?string;
+  parameters: Parameter[];
+  parametersByName: { [string]: Parameter };
+  originalParameters: Parameter[];
+  preparing: boolean;
+  canceled: boolean;
+  paused: boolean;
+  userCallback: CompletionCallback;
+  handle: ?any; // TODO: Figure out the type here.
+  error: ?Error;
+  connection: ?any; // TODO: This should be `Connection`, not `any`.
+
+  callback: () => void;
+
+  constructor(sqlTextOrProcedure: ?string, callback: CompletionCallback) {
     super();
 
     this.sqlTextOrProcedure = sqlTextOrProcedure;
     this.parameters = [];
     this.parametersByName = {};
+    this.originalParameters = [];
+    this.preparing = false;
+    this.handle = undefined;
     this.canceled = false;
     this.paused = false;
+    this.error = undefined;
+    this.connection = undefined;
     this.userCallback = callback;
     this.callback = function() {
       if (this.preparing) {
         this.emit('prepared');
-        return this.preparing = false;
+        this.preparing = false;
```

Comment:
I removed the `return` statements here, because I don't think these make sense, really, and they conflict the `void` return type of the `CompletionCallback`.